judge/views/two_factor.py

```diff
@@ -54,6 +54,10 @@ def get(self, request, *args, **kwargs):
         if not profile.totp_key:
             profile.totp_key = pyotp.random_base32(length=32)
             profile.save()
+        if not profile.scratch_codes:
+            profile.scratch_codes = json.dumps([pyotp.random_base32(length=16)
+                                                for i in range(settings.DMOJ_SCRATCH_CODES_COUNT)])
+            profile.save()
```

Comment:
Not sure if we should be saving the profile twice (once for the `totp_key` and again for the `scratch_codes`). Maybe have a list of `update_fields` and call `profile.save(update_fields=update_fields)` (since having an empty list automatically skips the save).

There might be a better way to do this that I can't think of.